docs/learn/streams.md

```diff
@@ -120,21 +178,106 @@ except OpenAIError as e:
+stream = recommend_book("fantasy")
+# Your IDE will provide autocompletion for stream methods and properties
+```
 
-Your IDE will recognize the response object when streaming as a provider-specific `BaseStream` instance and provide appropriate autocompletion and type checking for its methods and properties, improving your development experience when working with streamed LLM responses.
+This type safety ensures that your IDE can provide appropriate autocompletion and type checking for the stream's methods and properties, improving your development experience when working with streamed LLM responses.
 
 ## Best Practices
 
-- Real-time Feedback: Use streaming for applications where users benefit from seeing results immediately, such as chatbots or writing assistants.
-- Progress Indicators: Implement progress bars or loading animations that update based on the streamed response, improving user experience for longer generations.
-- Incremental Processing: Process streamed content incrementally for large outputs, reducing memory usage and allowing for early termination if needed.
-- Timeout Handling: Implement timeouts for streamed responses to handle cases where the LLM might take too long to generate content.
+When working with streaming in Mirascope, consider the following best practices:
+
+1. **Real-time Feedback**: Use streaming for applications where users benefit from seeing results immediately, such as chatbots or writing assistants.
+
+   ```python
+   @openai.call(model="gpt-4o-mini", stream=True)
+   @prompt_template("Write a short story about {topic}")
+   def write_story(topic: str):
+       ...
+
+   print("Generating story...")
+   for chunk, _ in write_story("a magical forest"):
+       print(chunk.content, end="", flush=True)
+   # > Generating story...
+   # Once upon a time, in a magical forest...
+   ```
+
+2. **Progress Indicators**: Implement progress bars or loading animations that update based on the streamed response, improving user experience for longer generations.
+
+   ```python
+   from tqdm import tqdm
+
+   @openai.call(model="gpt-4o-mini", stream=True)
+   @prompt_template("Summarize the following text: {text}")
+   def summarize_text(text: str):
+       ...
+
+   text = "..." # Long text to summarize
+   with tqdm(total=100, desc="Summarizing") as pbar:
+       for i, (chunk, _) in enumerate(summarize_text(text)):
+           print(chunk.content, end="", flush=True)
+           pbar.update(1)  # Update progress bar
+   ```
+
+3. **Incremental Processing**: Process streamed content incrementally for large outputs, reducing memory usage and allowing for early termination if needed.
+
+   ```python
+   @openai.call(model="gpt-4o-mini", stream=True)
+   @prompt_template("Generate a list of {n} random words")
+   def generate_words(n: int):
+       ...
+
+   word_count = 0
+   for chunk, _ in generate_words(1000):
+       words = chunk.content.split()
```

Comment:
what happens when there is a long word and it's never split? seems like we would increate the word count by `1` each time but we would still be on the same word as it streams?